app/assets/javascripts/application.js

```diff
@@ -375,7 +375,7 @@ $(function() {
 
 function update_puppetclasses(element) {
   var host_id = $("form").data('id')
-  var env_id = $('*[id*=environment_id]').val();
+  var env_id = $('*[name*=environment_id]').val();
```

Comment:
to make a long story short:
before select2 we had one select with id and name, e.g.,
```html```
<select id=host_env_id name=host[env_id]....
```
After select2 we have the select and also a div with a similar id:
```html```
<select id=host_env_id name=host[env_id]....
<div id=s2id_host_env_id....
```
So, when searching by ~id ```$('*[id*=environment_id]')``` it would yield both elements and it can't extract the value. But, the select element also has the name attribute (```name=host[environment_id]```) which is unique to that field - so I am hooking up to it.